scripts/triggered_mails.py

```diff
@@ -33,19 +32,13 @@ def main(dry_run=True):
 
 
 def find_inactive_users_with_no_inactivity_email_sent_or_queued():
-    inactive_users = User.find(
-        (Q('date_last_login', 'lt', timezone.now() - settings.NO_LOGIN_WAIT_TIME) & Q('tags__name', 'ne', 'osf4m')) |
-        (Q('date_last_login', 'lt', timezone.now() - settings.NO_LOGIN_OSF4M_WAIT_TIME) & Q('tags__name', 'eq', 'osf4m'))
-    )
-    inactive_emails = QueuedMail.find(Q('email_type', 'eq', NO_LOGIN_TYPE))
-
-    #This is done to prevent User query returns comparison to User, as equality fails
-    #on datetime fields due to pymongo rounding. Instead here _id is compared.
-    users_sent_id = [email.user._id for email in inactive_emails]
-    inactive_ids = [user._id for user in inactive_users if user.is_active]
-    users_to_send = [User.load(id) for id in (set(inactive_ids) - set(users_sent_id))]
-    return users_to_send
-
+    users_sent_ids = QueuedMail.objects.filter(email_type=NO_LOGIN_TYPE).values_list('user__guids___id')
```

Comment:
Having read the "Performance Optimizations" section of https://docs.djangoproject.com/en/1.11/ref/models/querysets/#in, I should probably change the exclude to be: `.exclude(guids___id__in=list(users_sent_ids))`